lib/util/apis.dart

```diff
@@ -653,4 +653,29 @@ Future<Map<String, dynamic>> changeProfilePicture(BuildContext context, String i
       }
     },
   );
+}
+
+Future<Map<String, dynamic>> finishOrExtend(BuildContext context, String goalId, String goalStatus, String goalExtension) async {
```

Comment:
im assuming goalStatus relates to isFinished, you can just do bool isFinished here

Reply:
same thing, I'm using query string param for goal status (I can rename it to isFinished tho)

Reply:
Yeah lets rename it to isFinished so we know that it relates to being a bool

Reply:
changed goalStatus to isFinished